Source/Core/Core/Core.cpp

```diff
@@ -146,7 +154,7 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	return s_cpu_thread.joinable() ? tls_is_cpu_thread : !s_is_started;
```

Comment:
I actually tested that for quite a while, and originally tried doing it without the joined() call, but for some reason IsCPUThread() is getting called from Source/Core/Core/HW/ProcessorInterface.cpp BEFORE the tls_is_cpu_thread variable gets set to true, but for some reason the program needed IsCPUThread to be returning true at this point. I'm not sure why though, as the CPUThread couldn't be calling IsCPUThread before it begins the CpuThread method.